osu.Game/Overlays/Music/PlaylistOverlay.cs

```diff
@@ -148,7 +149,16 @@ public void PlayNext()
         private void playSpecified(BeatmapInfo info)
         {
             beatmapBacking.Value = beatmaps.GetWorkingBeatmap(info, beatmapBacking);
-            beatmapBacking.Value.Track.Restart();
+
+            var track = beatmapBacking.Value.Track;
+
+            track.Restart();
+
+            // this is temporary until we have blocking (async.Wait()) audio component methods.
+            // then we can call RestartAsync().Wait() or the blocking version above.
+            while (!track.IsRunning)
+                Thread.Sleep(1);
+        }

```

Comment:
Can you make a common method for this stuff? It's used in both Player and here now.

Reply:
how would you like it to be done? they are both doing different things (one is waiting for running and one is waiting for stopped).

i'd rather just fixed the comment part than make a method for this.

Reply:
How about something like a `void WaitWhile(Predicate<bool> predicate)`? It would be easier to find the cases where this stuff is done when we finally have async methods, by just simply finding references to `WaitWhile`.

Reply:
Let's just track this task here, as there's only two custom usages like this (and I have nowhere to put this `WaitWhile` helper method) https://github.com/ppy/osu-framework/issues/1230